### PR TITLE
DAH-1659: Rent a percent of income

### PR DESCRIPTION
**Pull request by chadbrokaw-exygy**
https://sfgovdt.jira.com/browse/DAH-1659

### CODE REVIEW

File: app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx [hunk 1]
```diff
@@ -213,7 +223,9 @@ const buildAccordions = (
                     <div>
                       {renderInlineMarkup(
                         t("listings.incomeRange.minMaxPerMonth", {
-                          min: Math.round(occupancy?.absoluteMinIncome).toLocaleString(),
+                          min: Math.round(
+                            forceZeroInRange ? 0 : occupancy?.absoluteMinIncome
+                          ).toLocaleString(),
```
Comment (jimlin-sfgov) [inline app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx:228]: With the nested conditional, it's getting hard to see what the intended behavior is. Just want to double check that this is the the behavior for each scenario:
1. `occupancy?.absoluteMinIncome <= 0 && forceZeroInRange`
    - "$0 to $X per month"
2. `occupancy?.absoluteMinIncome <= 0 && !forceZeroInRange`
    - "up to $X per month"
3. `occupancy?.absoluteMinIncome > 0 && forceZeroInRange`
    - "$0 to $X per month"
4. `occupancy?.absoluteMinIncome > 0 && !forceZeroInRange`
    - "$Y to $X per month"

There's probably some way to express all the scenarios explicitly, while also keeping the code compact, but I haven't figured one out yet.

File: app/javascript/modules/listings/DirectoryHelpers.tsx [hunk 2]
```diff
@@ -71,15 +71,21 @@ const isNumber = (val: number) => val || val === 0
 // Transforms two numbers into a range string with optional currency formatting and optional suffix
 // 100, 200, true, null --> $100 - $200
 // 100, 100, false, "%" --> 100%
```
Comment (jimlin-sfgov) [inline app/javascript/modules/listings/DirectoryHelpers.tsx:73]: With the new `forceZeroInRange` argument introduced, existing comments should be updated.

**Review by jimlin-sfgov — CHANGES_REQUESTED** [review]

File: app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx (hunk 1 above)
Comment (chadbrokaw-exygy) [reply app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx:228]: Yes, that is correct

**Review by jimlin-sfgov — APPROVED** [review]
looks good!

**Pull request merged**